agent/engine/dependencygraph/graph.go

```diff
@@ -202,6 +211,21 @@ func verifyContainerDependenciesResolved(target *api.Container, existingContaine
 	return true
 }
 
+func verifyResourceDependenciesResolved(target *api.Container, existingResources map[string]taskresource.TaskResource) bool {
+	targetNext := target.GetNextKnownStateProgression()
+	resourceDependencies := target.TransitionDependenciesMap[targetNext].ResourceDependencies
+	for _, resourceDependency := range resourceDependencies {
+		dep, exists := existingResources[resourceDependency.ResourceName]
+		if !exists {
+			return false
+		}
+		if dep.GetKnownStatus() < resourceDependency.SatisfiedStatus {
```

Comment:
please use a getter here (and everywhere else) instead of accessing the field directly.